test/util/diff.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/pmezard/go-difflib/difflib"
+)
+
+// Refresh controls whether to update the golden artifacts instead.
+// It is set using the environment variable REFRESH_GOLDEN.
+func Refresh() bool {
+	v, exists := os.LookupEnv("REFRESH_GOLDEN")
```

Comment:
It's possible but I don't it's worth explaining how to do it with bazel.  On the other hand `env REFRESH_GOLDEN=true go test ./...` just works.